module/os/linux/spl/spl-kmem-cache.c

```diff
@@ -160,16 +160,32 @@ taskq_t *spl_kmem_cache_taskq;		/* Task queue for aging / reclaim */
 
 static void spl_cache_shrink(spl_kmem_cache_t *skc, void *obj);
 
+typedef struct {
+	uint32_t magic;
```

Comment:
All the magic-marking may be overkill - I wanted good diagnostics when this falls over, but it seems stable and good.
Opinions?